app/src/main/java/com/kickstarter/ui/fragments/PledgeFragment.kt

```diff
@@ -0,0 +1,275 @@
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        cards_recycler?.adapter = null
+    }
+
+    override fun closePledgeButtonClicked(position: Int) {
+        this.viewModel.inputs.closeCardButtonClicked(position)
+    }
+
+    override fun pledgeButtonClicked(viewHolder: RewardPledgeCardViewHolder) {
+        this.viewModel.inputs.pledgeButtonClicked()
+    }
+
+    override fun selectCardButtonClicked(position: Int) {
+        this.viewModel.inputs.selectCardButtonClicked(position)
+    }
+
+    override fun addNewCardButtonClicked() {
+        this.viewModel.inputs.newCardButtonClicked()
+    }
+
+    private fun updatePledgeCardSelection(positionAndSelected: Pair<Int, Boolean>) {
+        val position = positionAndSelected.first
+        val selected = positionAndSelected.second
+        if (selected) {
+            (cards_recycler.adapter as RewardCardAdapter).setSelectedPosition(position)
+            cards_recycler.scrollToPosition(position)
+        } else {
+            (cards_recycler.adapter as RewardCardAdapter).resetSelectedPosition(position)
+        }
+        (cards_recycler.layoutManager as FreezeLinearLayoutManager).setFrozen(selected)
+    }
+
+    private fun showPledgeSection(rewardAndLocation: Pair<Reward, ScreenLocation>) {
+        val location = rewardAndLocation.second
+        val reward = rewardAndLocation.first
+        setInitialViewStates(location, reward)
+        revealPledgeSection(location)
+    }
+
+    private fun setInitialViewStates(location: ScreenLocation, reward: Reward) {
+        positionRewardSnapshot(location, reward)
+
+        setDeliveryHeight(location)
+
+        pledge_details.y = pledge_root.height.toFloat()
+    }
+
+    private fun setDeliveryHeight(location: ScreenLocation) {
+        val miniRewardWidth = this.resources.getDimensionPixelSize(R.dimen.mini_reward_width)
+        val scaleX = miniRewardWidth.toFloat() / location.width
+        val miniRewardHeight = location.height * scaleX
+        delivery.measure(View.MeasureSpec.makeMeasureSpec(pledge_details.width, View.MeasureSpec.UNSPECIFIED),
+                View.MeasureSpec.makeMeasureSpec(pledge_details.height, View.MeasureSpec.UNSPECIFIED))
+        val targetHeight = delivery.measuredHeight
+        val deliveryParams = delivery.layoutParams as LinearLayout.LayoutParams
+        deliveryParams.height = Math.max(miniRewardHeight.toInt(), targetHeight)
+        delivery.layoutParams = deliveryParams
+    }
+
+    private fun positionRewardSnapshot(location: ScreenLocation, reward: Reward) {
+        val rewardParams = reward_snapshot.layoutParams as FrameLayout.LayoutParams
+        rewardParams.marginStart = location.x.toInt()
+        rewardParams.topMargin = location.y.toInt()
+        rewardParams.height = location.height
+        rewardParams.width = location.width
+        reward_snapshot.layoutParams = rewardParams
+        reward_snapshot.pivotX = 0f
+        reward_snapshot.pivotY = 0f
+
```

Comment:
In the description of the PR, I say I am purposely setting it to 0, 0 because we don't have the real rewards 😅  But this point is moot because I changed it in my refactor.